### PR TITLE
Update major libs dep fix #398

### LINKED ISSUE #398: New semver-major release of istanbul libraries
I just wanted to notify you that I'm in the process of a semver-major release of the istanbul libraries.  They are currently published to npm under the `next` tag but will likely be promoted to `latest` soon.  The breaking changes include:
* dropping of node.js 6 (8.9.0 is the oldest officially support version)
* some API changes (some functions are now async, changes to reporting)

I don't have time right now to create a PR for this but if you do I can try giving pointers if you run into issues (feel free to tag me in any PR).

### PR DESCRIPTION
**Pull request by anthony-redFox**

### CODE REVIEW

**Comment by anthony-redFox** [comment]
@coreyfarrell and @johnjbarton please take a look

File: lib/reporter.js [hunk 1]
```diff
@@ -214,28 +217,24 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
...
         }
 
-        var mainDir = reporterConfig.dir || config.dir
-        var subDir = reporterConfig.subdir || config.subdir
-        var browserName = browser.name.replace(':', '')
-        var simpleOutputDir = generateOutputDir(browserName, mainDir, subDir)
-        var resolvedOutputDir = path.resolve(basePath, simpleOutputDir)
+        const mainDir = reporterConfig.dir || config.dir
+        const subDir = reporterConfig.subdir || config.subdir
+        const simpleOutputPath = generateOutputPath(basePath, browser.name, mainDir, subDir)
```
Comment (johnjbarton) [inline lib/reporter.js]: what makes this "simple"? Wouldn't `outputPath` be clear?

Comment (anthony-redFox) [reply lib/reporter.js]: done. thanks!

File: lib/reporter.js [hunk 2]
```diff
@@ -57,7 +49,7 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
 
   function normalize (key) {
     // Exclude keys will always be relative, but covObj keys can be absolute or relative
-    var excludeKey = isAbsolute(key) ? path.relative(basePath, key) : key
+    var excludeKey = path.isAbsolute(key) ? path.relative(basePath, key) : key
```
Comment (johnjbarton) [inline lib/reporter.js]: AFAICT this commit is not related to major libs deps. Please move to separate Pull Request.

File: lib/reporter.js [hunk 3]
```diff
@@ -245,31 +244,29 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
...
+          const coverageFailed = checkCoverage(browser, remappedCoverageMap)
+          if (coverageFailed && results) {
+            results.exitCode = 1
           }
         }
 
-        var context = istanbulLibReport.createContext(options)
-        var report = reports.create(reporterConfig.type || 'html', options)
+        const context = istanbulLibReport.createContext(options)
+        const report = reports.create(reporterConfig.type || 'html', options)
```
Comment (johnjbarton) [inline lib/reporter.js]: Sorry I can't figure out what this commit is achieving.  I  *think* it is:
```
fix(reporter): update calls to  match new API in istanbul-lib-report v?
```

File: lib/reporter.js (hunk 2 above)
Comment (anthony-redFox) [reply lib/reporter.js]: Done

File: lib/reporter.js (hunk 3 above)
Comment (anthony-redFox) [reply lib/reporter.js]: I split refactoring and upgrade on separate PRs

**Comment by anthony-redFox** [comment]
I will do. Thanks

File: lib/reporter.js [hunk 4]
```diff
@@ -218,11 +218,11 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
     coverageMap.merge(result.coverage)
   }
 
-  this.onRunComplete = function (browsers, results) {
+  this.onRunComplete = async function (browsers, results) {
```
Comment (matz3) [inline lib/reporter.js:221]: Having this function async causes `result.exitCode` to be set too late to be taken into account when karma exits the process.
So when the coverage check thresholds are not met, the process still exits with code 0.

Comment (matz3) [reply lib/reporter.js:221]: I've created https://github.com/karma-runner/karma-coverage/issues/418 to track this issue
